babyjub/eddsa.go

```diff
@@ -260,3 +283,27 @@ func (p *PublicKey) VerifyPoseidon(msg *big.Int, sig *Signature) bool {
 	right.Add(sig.R8, right) // right = 8 * R + 8 * hm * A
 	return (left.X.Cmp(right.X) == 0) && (left.Y.Cmp(right.Y) == 0)
 }
+
+// Scan implements Scanner for database/sql.
+func (p *PublicKey) Scan(src interface{}) error {
+	srcB, ok := src.([]byte)
+	if !ok {
+		return fmt.Errorf("can't scan %T into PublicKey", src)
+	}
+	var comp PublicKeyComp
+	if err := comp.UnmarshalText(srcB); err != nil {
+		return err
+	}
+	decomp, err := comp.Decompress()
+	if err != nil {
+		return err
+	}
+	*p = *decomp
+	return nil
+}
+
+// Value implements valuer for database/sql.
+func (p *PublicKey) Value() (driver.Value, error) {
+	comp, err := p.Compress().MarshalText()
+	return comp, err
+}
```

Comment:
Suggestion to simplify and work with bytes like in Signature (instead of string taken as bytes)
```suggestion
// Scan implements Scanner for database/sql.
func (p *PublicKey) Scan(src interface{}) error {
	srcB, ok := src.([]byte)
	if !ok {
		return fmt.Errorf("can't scan %T into PublicKey", src)
	}
	var comp PublicKeyComp
	copy(comp[:], srcB)
	decomp, err := comp.Decompress()
	if err != nil {
		return err
	}
	*p = *decomp
	return nil
}

// Value implements valuer for database/sql.
func (p *PublicKey) Value() (driver.Value, error) {
	comp := p.Compress()
	return comp[:], nil
}
```